# Problem 8 #

### a) We need to find x(v) ###


$$ a(v) = a_0(1- \frac{v^2}{v_m^2}) $$

$$ a=\frac{dv}{dt} , v=\frac{dx}{dt}, a=v\frac{dv}{dx} $$

$$ v\frac{dv}{dx}=a_0(1-\frac{v^2}{v_m^2}) $$

$$ \frac{vdv}{(1-\frac{v^2}{v_m^2})} = a_0dx $$

$$ vdv / (\frac{v_m^2-v^2}{v_m^2}) = a_0dx $$

$$ \frac{v_m^2vdv}{v_m^2-v^2} = a_0dx $$

$$ v_m^2\int{\frac{vdv}{v_m^2-v^2}} = a_0\int{dx} $$

$$ u = -v^2, \frac{du}{dv} = -2v $$

$$ -\frac{v_m^2}{2} \int{\frac{du}{v_m^2-u}} = a_0\int{dx} $$

$$ -\frac{v_m^2}{2}(ln{|v_m^2-v_f^2|} - ln {|v_m^2-v_0^2 |} ) = a_0(x_1 - x_0) $$

$$ -\frac{v_m^2}{2}(ln{|\frac{v_m^2-v_f^2}{v_m^2}|} ) = a_0(x_1 - x_0) $$

$$ x_1 = -(\frac{v_m^2}{2}(ln{|\frac{v_m^2-v_f^2}{v_m^2}|} ))/ a_0 + x_0 $$

In [1]:
from sympy import *
import sympy.physics.units as u
init_printing(use_latex='mathjax')
#Constants
knots = 1.852/3.6
vm_c = 140 * knots
a0_c = 1.5
#symbols
x1 = Symbol("x_1")

def x_1(vf, x_0=0):
    res = -(vm_c**2/2 * log((vm_c**2-vf**2)/vm_c**2)) /a0_c + x_0
    return res
res_a = x_1(60 * knots)

Eq(x1, round(res_a, 2)*u.m)

x₁ = 350.9⋅m

### b) Compute the total distance travelled when reaching the height of h =  50 ft ###

$$ x_t = x_1 + x_2 $$

Time to travel to height 50 
$$ y_2 = y_1 +wt_2 $$
(1)
$$ t_2 = \frac{y_2}{w} $$

After taking off velocity in the x direction is constant as velocity in the y direction is constant, this relationship is important to realize otherwise things get really messy.

$$ \int_{x_1=0}^{x_2}{dx}= v_1\int_{t_1=0}^{t_2}{dt} $$
$$ x_2 = v_1t_2 $$

put (1) into above equation
$$x_2(t) = v_1\frac{h}{w}$$

$$ x_t = 350.9 +\frac{50*60*\frac{1.852}{3.6}}{{1000}}$$

In [2]:
w_c = 1000
h_c = 50
v1_c = 60 * knots
v1, h, w, x2, xt = symbols("v_1 h w x_2 x_t")

eq = Eq(x2, v1*(h/w))
eq = eq.subs(v1, v1_c,)
eq = eq.subs(h, h_c)
eq = eq.subs(w, w_c)
Eq(xt, round(res_a + eq.args[1], 1)*u.m, )

x_t = 352.4⋅m